### PR TITLE
feat(ica): allow unordered ica channels

### PR DESCRIPTION
**Pull request by srdtrk**
## Description

This PR can be reviewed now. But some small PRs might still be merged in before this is merged to main

closes: #3996

### Commit Message / Changelog Entry

```text
feat(ica): allow unordered ica channels
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
Great work team!! :mechanical_arm:

**Comment by colin-axner** [comment]
Looks like the e2e is failing:
```
=== NAME  TestInterchainAccountsTestSuite/TestMsgSendTx_SuccessfulTransfer_AfterUpgradingOrdertoUnordered
    base_test.go:550: 
        	Error Trace:	/home/runner/work/ibc-go/ibc-go/e2e/tests/interchain_accounts/base_test.go:550
        	Error:      	Received unexpected error:
        	            	failed waiting for condition after 120.000000 seconds
        	Test:       	TestInterchainAccountsTestSuite/TestMsgSendTx_SuccessfulTransfer_AfterUpgradingOrdertoUnordered
        	Messages:   	channel was not upgraded: expected ORDER_UNORDERED got ORDER_ORDERED
```

**Comment by colin-axner** [comment]
We decided to comment out the failing e2e for now and uncomment it when a fix is pushed in hermes

**Review by DimitrisJim — APPROVED** [review]
:rocket: lgtm!

File: e2e/tests/interchain_accounts/base_test.go [hunk 1]
```diff
@@ -406,3 +424,132 @@ func (s *InterchainAccountsTestSuite) TestMsgSendTx_SuccessfulTransfer_AfterReop
 		s.Require().Equal(expected, balance.Int64())
 	})
 }
+
+/*
+TOOD: uncomment when hermes works with upgrades, https://github.com/cosmos/ibc-go/issues/5644
```
Comment (damiannolan) [inline e2e/tests/interchain_accounts/base_test.go:429]: 👍🏻

**Review by damiannolan — APPROVED** [review]
LGTM!

**Comment by DimitrisJim** [comment]
merging to start working on the bp.

**Pull request merged**